src/ml/neural_net/mps_layer_helper.h

```diff
@@ -0,0 +1,41 @@
+#import <Accelerate/Accelerate.h>
+#import <Foundation/Foundation.h>
+#import <MetalPerformanceShaders/MetalPerformanceShaders.h>
+#import <Metal/Metal.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+API_AVAILABLE(macos(10.14))
+@interface TCMPSLayerHelper: NSObject
```

Comment:
It looks like you're creating a class just to define some free functions in Objective C. The more typical way of doing this might be to add a category to e.g MPSCNNFullyConnectedNode, something like

@interface MPSCNNFullyConnectNode (TCMPSLayerHelper)

+ (instancetype)fullyConnectNodeWithInput:...

@end

Reply:
done!